src/character.cpp

```diff
@@ -707,9 +707,62 @@ double Character::aim_speed_dex_modifier() const
     return get_dex() - 8;
 }
 
-double Character::aim_speed_encumbrance_modifier() const
+float Character::aim_speed_modifier() const
 {
-    return ( encumb( body_part_hand_l ) + encumb( body_part_hand_r ) ) / 10.0;
+    return manipulator_score();
+}
+
+float Character::melee_thrown_move_modifier_hands() const
+{
+    if( manipulator_score() == 0.0f ) {
+        return MAX_MOVECOST_MODIFIER;
+    } else {
+        return std::min( MAX_MOVECOST_MODIFIER, 1.0f / manipulator_score() );
+    }
+}
+
+float Character::melee_thrown_move_modifier_torso() const
+{
+    const bodypart *torso = get_part( body_part_torso );
+    // TODO: make this "balance" perhaps?
+    if( torso == nullptr || torso->encumb_adjusted_limb_value( 1.0f ) == 0.0f ) {
+        return MAX_MOVECOST_MODIFIER;
+    } else {
+        return std::min( MAX_MOVECOST_MODIFIER, 1.0f / torso->encumb_adjusted_limb_value( 1.0f ) );
+    }
+}
+
+float Character::melee_stamina_cost_modifier() const
+{
+    if( lifting_score( body_part_type::type::arm ) == 0.0f ) {
+        return MAX_MOVECOST_MODIFIER;
+    } else {
+        return std::min( MAX_MOVECOST_MODIFIER, 1.0f / lifting_score( body_part_type::type::arm ) );
+    }
+}
+
+float Character::reloading_move_modifier() const
+{
+    if( manipulator_score() == 0.0f ) {
+        return MAX_MOVECOST_MODIFIER;
+    } else {
+        return std::min( MAX_MOVECOST_MODIFIER, 1.0f / manipulator_score() );
+    }
+}
+
+float Character::thrown_dex_modifier() const
+{
+    return manipulator_score();
+}
+
+float Character::ranged_dispersion_modifier() const
+{
+    if( lifting_score( body_part_type::type::arm ) == 0.0f ) {
+        return MAX_MOVECOST_MODIFIER;
+    } else {
+        return std::min( MAX_MOVECOST_MODIFIER,
+                         ( 10.0f / lifting_score( body_part_type::type::arm ) ) - 10 );
+    }
```

Comment:
1) well the more scores you add the worse the thing is, unless you use some kind of weighted score. i just made a judgement call based on what i think would be used to aim a gun.
3) for a lower score, i needed a higher output value. it also needed to scale a bit more to actually affect the dispersion how it was affected before, which was `encumbrance / 5`. the 10 value is subject to change based on balance

Reply:
`1.`: Point; for multipliers, square root? I can see lifting strength possibly being primary for a longarm, but I'm dubious about something the size of a pistol. sqrt( manipulation_multiplier * lifting_score_multiplier ) would be a compromise. EDIT: The problem being that this isn't actually a multiplier; see below.
`3.` From the ranged_balance_test, a bit of adjustment is likely needed, unless the test is considered too lenient.

After having looked over it again, I realized that it isn't acting as a (multiplicative) modifier in the first place. (It may be clearer not to call it a modifier if it not acting like a multiplicative modifier is desired.) This makes it even odder to use MAX_MOVECOST_MODIFIER. I'm also uncertain on why you're substituting something not modified by encumbrance for, well, encumbrance.